client/src/pages/HomePage.js

```diff
@@ -0,0 +1,169 @@
+import React, { Component } from "react";
+import axios from "axios";
+import { ChevronRightIcon } from "@modulz/radix-icons";
+
+import AuthService from "../services/auth.service";
+import Avatar from "../components/Avatar";
+import { Link } from "react-router-dom";
+import Plural from "../components/Plural";
+import FightPilette from "../images/fight.png";
+import WaitPilette from "../images/attente.png";
+
+class HomePage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentUser: null,
+      usersData: [],
+      toPlayChallenges: [],
+      pendingChallenges: [],
+    };
+  }
+
+  componentDidMount() {
+    axios
+      .get("/api/v1/duels/")
+      .then((res) => {
+        const toPlay = [],
+          pending = [];
+        res.data.forEach((val) => {
+          if (!val.inProgress) return;
+          if (val.rounds[val.currentRound - 1][0].userAnswer !== undefined) {
+            pending.push(val);
+          } else {
+            toPlay.push(val);
+          }
+        });
+        this.setState({
+          toPlayChallenges: toPlay,
+          pendingChallenges: pending,
+        });
+
+        const listOfUsers = [
+          ...toPlay.map((value) => value.opponent),
+          ...pending.map((value) => value.opponent),
+        ];
+        this.getUsersData(listOfUsers);
+      })
+      .catch((err) => console.error(err));
+  }
+
+  getUsersData(otherUsers) {
+    const currentUser = AuthService.getCurrentUser();
+    const listOfUsers = [currentUser.pseudo, ...otherUsers];
+    axios
+      .post("/api/v1/users/", listOfUsers)
+      .then((res) => {
+        this.setState({
+          currentUser: res.data[currentUser.pseudo],
+          usersData: res.data,
+        });
+      })
+      .catch((err) => console.error(err));
+  }
+
+  render() {
+    const { currentUser, usersData, toPlayChallenges, pendingChallenges } = this.state;
+    const cuWins = currentUser?.wins ?? 0;
+    const cuLosses = currentUser?.losses ?? 0;
+
+    return (
+      <main id="homepage">
+        <header>
+          <div id="header-background"></div>
+          <Avatar
+            size="125px"
+            eyes={currentUser?.avatar.eyes ?? 0}
+            hands={currentUser?.avatar.hands ?? 0}
+            hat={currentUser?.avatar.hat ?? 0}
+            mouth={currentUser?.avatar.mouth ?? 0}
+            colorBG={currentUser?.avatar.colorBG ?? "#d3d3d3"}
+            colorBody={currentUser?.avatar.colorBody ?? "#0c04fc"}
```

Comment:
If the props are required, I think they should not, and the `Avatar` should handle itself the default values

Reply:
Yes!